subsys/net/lib/nrf_cloud/Kconfig

```
@@ -1,8 +1,23 @@
-# Copyright (c) 2018 Nordic Semiconductor
+# Copyright (c) 2018 - 2025 Nordic Semiconductor
 #
 # SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
 #
 
+config SUPPORT_NRF_CLOUD
+	bool
+	depends on SOC_SERIES_NRF91X || (SOC_SERIES_NRF53X && WIFI_NRF7002)
+	default y
+	help
+	  Hidden option to indicate if there is support for nRF cloud
+
+config NRF_CLOUD
+	bool "nRF Cloud support"
+	depends on SUPPORT_NRF_CLOUD
+	help
+	  Enable configuration options for nRF Cloud.
+
+if NRF_CLOUD
```

```suggestion
if SUPPORT_NRF_CLOUD
```

If you want to block nrfCloud use from certain devices, you can just hide all these using the `SUPPORT_NRF_CLOUD`. You don't  need that separate config variable that needs to be turned on.

---

True. But if you think about other libraries, you quite normally have a "main switch" where you enable the library manually and then set other Kconfigs related to library. We don't need to add all those Kconfig symbols into samples that don't use NRF_CLOUD at all even in nrf91 or nrf53+wifi.

But I understand if your suggested change is done to keep the change backwards compatible.